domain/src/Nft/Reactors/NftReactor.php

```diff
@@ -20,15 +21,22 @@ public function __construct(private TaxYearRepository $taxYearRepository)
 
     public function handleNftDisposedOf(NftDisposedOf $event, Message $message): void
     {
-        $taxYearId = TaxYearId::fromYear($event->date->getYear());
-        $taxYear = $this->taxYearRepository->get($taxYearId);
+        $taxYear = TaxYearGenerator::fromYear($event->date->getYear());
+        $taxYearId = TaxYearId::fromTaxYear($taxYear);
+        $taxYearAggregate = $this->taxYearRepository->get($taxYearId);
 
         if ($event->proceeds->isGreaterThan($event->costBasis)) {
-            $taxYear->recordCapitalGain(new RecordCapitalGain(amount: $event->proceeds->minus($event->costBasis)));
+            $taxYearAggregate->recordCapitalGain(new RecordCapitalGain(
+                taxYear: $taxYear,
+                amount: $event->proceeds->minus($event->costBasis),
+            ));
         } else {
-            $taxYear->recordCapitalLoss(new RecordCapitalLoss(amount: $event->costBasis->minus($event->proceeds)));
+            $taxYearAggregate->recordCapitalLoss(new RecordCapitalLoss(
+                taxYear: $taxYear,
+                amount: $event->costBasis->minus($event->proceeds),
+            ));
         }
```

Comment:
ℹ️ Now passing the tax year to the actions (same for the share pooling ones), so it can be passed on to the events and ultimately, recorded in the projection.

I'm not very happy with that – the `TaxYear` aggregate root should probably hold this value itself, instead of relying on the actions.